### PR TITLE
enable promtail deployment on WCs

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/roadmap/issues/2761

Tested on a v19 WC and promtail is deployed but unable to send logs to loki on the MC. Need to solve this. 
Also gonna test it on a v18.4.2 WC.

### CODE REVIEW

File: internal/controller/vintagewc_controller.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func (r *VintageWCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 
-	logger.Info("Reconciling Vintage WV cluster", "name", cluster.GetName())
+	logger.Info("Reconciling Vintage WC cluster", "name", cluster.GetName())
```
Comment (QuentinBisson) [inline internal/controller/vintagewc_controller.go:67]: 👍🏻

File: pkg/resource/promtail-wiring/reconciler.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ type Reconciler struct {
 // ReconcileCreate ensure user value configmap is set in observability bundle
 // for the given cluster.
 func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
+	isWC := lc.GetClusterName() != lc.GetInstallationName()
```
Comment (QuentinBisson) [inline pkg/resource/promtail-wiring/reconciler.go:31]: Can you check @marieroque PR and do the same?

File: pkg/resource/promtail-wiring/reconciler.go [hunk 3]
```diff
@@ -42,16 +44,20 @@ func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Inter
...
 	}
 
-	// Set user value configmap in the app.
-	if setUserConfig(&currentApp, lc) {
-		logger.Info("promtailwiring updating")
-		// Update the app.
-		err := r.Client.Update(ctx, &currentApp)
-		if err != nil {
-			return ctrl.Result{}, errors.WithStack(err)
+	if !isWC {
```
Comment (QuentinBisson) [inline pkg/resource/promtail-wiring/reconciler.go:47]: I'd probably also check if this is vintage

File: pkg/vintagewc/cluster.go [hunk 4]
```diff
@@ -46,5 +50,5 @@ func (o Object) GetObject() client.Object {
 
 // on vintage WC, use extraconfig
 func (o Object) GetObservabilityBundleConfigMap() string {
-	return "observability-bundle-logging-values"
+	return "observability-bundle-logging-extraconfig"
```
Comment (QuentinBisson) [inline pkg/vintagewc/cluster.go:53]: Is there a reason to remame it?

File: pkg/resource/promtail-toggle/observability_bundle_configmap.go [hunk 5]
```diff
@@ -25,7 +25,7 @@ func ObservabilityBundleConfigMapMeta(lc loggedcluster.Interface) metav1.ObjectM
 		Namespace: lc.GetAppsNamespace(),
 		Labels: map[string]string{
 			// This label is used by cluster-operator to find extraconfig
-			"app.kubernetes.io/name": lc.AppConfigName("observability-bundle"),
+			"app.kubernetes.io/name": "observability-bundle",
```
Comment (QuentinBisson) [inline pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: That's true but not that is will only work on vintage WCs, Could you mention it in the comment?

File: pkg/vintagewc/cluster.go (hunk 4 above)
Comment (QuantumEnigmaa) [reply pkg/vintagewc/cluster.go:53]: No particular reason, just gives directly the information about the nature of the cm

File: pkg/resource/promtail-toggle/observability_bundle_configmap.go (hunk 5 above)
Comment (QuantumEnigmaa) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: Good point !

File: pkg/resource/promtail-wiring/reconciler.go (hunk 2 above)
Comment (QuantumEnigmaa) [reply pkg/resource/promtail-wiring/reconciler.go:31]: Sure :)

File: pkg/vintagewc/cluster.go (hunk 4 above)
Comment (QuentinBisson) [reply pkg/vintagewc/cluster.go:53]: But in that case that brings information about how it's mounted instead of what it is about :)

**Review by QuentinBisson — APPROVED** [review]
LGTM

File: pkg/resource/promtail-wiring/reconciler.go (hunk 2 above)
Comment (QuantumEnigmaa) [reply pkg/resource/promtail-wiring/reconciler.go]: Done

File: pkg/vintagewc/cluster.go (hunk 4 above)
Comment (QuantumEnigmaa) [reply pkg/vintagewc/cluster.go:53]: True but I guess it's less confusing for people to have really distinct name for those cms

Comment (QuentinBisson) [reply pkg/vintagewc/cluster.go:53]: Yep, but maybe we should name it with something related to promtail-toggle then but maybe we can discuss that later :)

File: pkg/resource/promtail-toggle/observability_bundle_configmap.go (hunk 5 above)
Comment (hervenicol) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: This is why we were using a function from the cluster type-specific package, so we return a different value depending on whether we're on CAPI or vintage.

Comment (QuentinBisson) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: Oh then I guess this implementation needs to be fixed

**Comment by QuantumEnigmaa** [comment]
Apparently CVEs were detected

**Comment by QuantumEnigmaa** [comment]
This is kind of troublesome : even the latest stable version (0.28.1) of the `k8s.io/apiextensions-apiserver` still has the same CVE as the one with our current verion (0.26.1)

**Comment by QuentinBisson** [comment]
Well if it does unless you want to fix it in apiextension-server upstream, I'd ignore the CVE with nancy, especially as we're not sure this will even work on 1.24 clusters

File: go.mod [hunk 6]
```diff
@@ -4,74 +4,75 @@ go 1.18
...
+	github.com/onsi/ginkgo/v2 v2.11.0
+	github.com/onsi/gomega v1.27.8
 	github.com/pkg/errors v0.9.1
-	k8s.io/api v0.26.1
-	k8s.io/apimachinery v0.26.1
-	k8s.io/client-go v0.26.1
-	sigs.k8s.io/cluster-api v1.4.2
-	sigs.k8s.io/controller-runtime v0.14.5
+	k8s.io/api v0.29.0-alpha.0
+	k8s.io/apimachinery v0.29.0-alpha.0
```
[the remark this answers is not in the document]
Comment (QuantumEnigmaa) [reply go.mod:11]: Is there a particular reason for it ?

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply go.mod:11]: kubernetes api is compatible up to 3 version and down to 2 versions. We run 1.24 so 1.29 would not be compatible with 1.24 ;)

[the remark this answers is not in the document]
Comment (QuantumEnigmaa) [reply go.mod:11]: Oh ok sure !

File: pkg/resource/promtail-toggle/observability_bundle_configmap.go (hunk 5 above)
Comment (hervenicol) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: I see this comment as "resolved", but yet we still have a hardcoded value that will not work in all situations.

Comment (QuentinBisson) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: Oh I was too eager Yesterday

Comment (QuantumEnigmaa) [reply pkg/resource/promtail-toggle/observability_bundle_configmap.go:28]: I am currently reworking on it

**Comment by QuantumEnigmaa** [comment]
```
r.LoggingReconciler.Reconcile: capicluster.Object does not implement loggedcluster.Interface (missing ObservabilityBundleConfigLabelName method)
```
Same for vintagemc and vintagewc but I guess we just need to update the dependency

**Comment by QuantumEnigmaa** [comment]
Tested and working :) 
I guess we're ready to merge

File: pkg/vintagewc/cluster.go [hunk 7]
```diff
@@ -25,7 +25,15 @@ func (o Object) GetAppsNamespace() string {
...
-	return app
+	if app == "observability-bundle" {
+		return o.GetObject().GetName() + "-" + app
+	} else {
+		return app
+	}
+}
+
+func (o Object) ObservabilityBundleConfigLabelName(config string) string {
+	return config
```
Comment (QuentinBisson) [inline pkg/vintagewc/cluster.go:36]: Should this not have some kind of logic ?

Comment (QuantumEnigmaa) [reply pkg/vintagewc/cluster.go:36]: The method above (`AppConfigName`) was doing exactly the same before I updated it to match the needs for the WCs so I guess this is fine

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**